### PR TITLE
feat: Better question submission flow

### PR DESCRIPTION
**Pull request by Akshay1018**
Added success popup on submitting a new question. 
Removed extra console.log() from the file
submitQuestion

### CODE REVIEW

File: src/components/QuestionSubmissionForm.tsx [hunk 1]
```diff
@@ -33,149 +49,206 @@ export function QuestionSubmissionForm({ session }: PageProps) {
...
+                leaveTo='opacity-0 scale-95'
+              >
+                <Dialog.Panel className='w-full max-w-md transform overflow-hidden rounded-2xl bg-white p-6 text-left align-middle shadow-xl transition-all'>
+                  <Dialog.Title as='h3' className='text-lg font-medium leading-6 text-gray-900'>
+                    Question submitted successfully!
+                  </Dialog.Title>
+                  <div className='mt-2'>
+                    <p className='text-sm text-gray-500'>
+                      Your question has been successfully submitted. We will approve your qustion shortly. Thanks for
+                      your contribution.
```
Comment (ykdojo) [inline src/components/QuestionSubmissionForm.tsx:233]:
```suggestion
                      Thank you! We will review the question as soon as possible.
```

**Review by ykdojo — CHANGES_REQUESTED** [review]

**Review by adilwahla — COMMENTED** [review]
👍

**Comment by ykdojo** [comment]
A thumbs up doesn't say much. What are the updates on this PR?

**Comment by ykdojo** [comment]
@Akshay1018 thank you so much!

I tested it on the deployed Vercel environment, and I'm getting an error:

image

Mind taking a look?

**Comment by Akshay1018** [comment]
@ykdojo Sure! I'm checking.

**Comment by Akshay1018** [comment]
@ykdojo If you don't mind could you please check the backend code? When we are getting the confirmation email after signing up and by clicking on confirm "Confirm email", it is taking us to localhost. Please have a look once.
Screenshot (18)
Screenshot (19)

**Comment by Akshay1018** [comment]
@ykdojo The issue is with the backend code because the column name is not matching in questions table. Please have a look at
the error. In the payload, we are sending "asked_date" and in the backend, it is expecting "estimated_asked_date". So that's why it is mismatching and throwing errors.
I want you to see line no. 62 in the screenshot and in the database please update it to "asked_date". The issue will be resolved.  Any further queries or errors please do let me know. Thanks.
Screenshot (20)
Screenshot (21)

**Comment by subhoghoshX** [comment]
That shouldn't be an issue as `estimated_asked_date` is being created [here](https://github.com/ykdojo/defaang/blob/main/src/pages/api/submit_question.ts#L28).

**Comment by Akshay1018** [comment]
@subhoghoshX If you don't mind could you please check the column name of the questions table in the SQL database? Check if it is `estimated_asked_date` or `asked_date`. The same thing happened when I was testing the local machine.

**Comment by Akshay1018** [comment]
@subhoghoshX check for this domain `https://defaang-qh3dsuwex-ykdojo.vercel.app/`.

**Comment by subhoghoshX** [comment]
I think I got it. Your branch is not up to date. Please sync your branch.

**Comment by subhoghoshX** [comment]
I think I got it. Your branch is not up to date. Please sync your branch. It should be fixed.

**Comment by Akshay1018** [comment]
@subhoghoshX exactly! I will pull the code. Thanks.

**Comment by Akshay1018** [comment]
@ykdojo @subhoghoshX Its's working with no errors this time. Please check from your end once. Thanks.
Screenshot (22)
Screenshot (23)

File: src/components/QuestionSubmissionForm.tsx [hunk 2]
```diff
@@ -33,149 +49,203 @@ export function QuestionSubmissionForm({ session }: PageProps) {
...
-
     reset();
   };
 
   return (
-    <div className='w-full rounded-md bg-white p-4 shadow-md sm:w-4/5 md:w-2/3 lg:w-1/2'>
-      <form className='space-y-8 divide-y divide-gray-200' onSubmit={handleSubmit(onSubmit)}>
-        <div className='space-y-8 divide-y divide-gray-200'>
-          <div>
+    <>
```
Comment (subhoghoshX) [inline src/components/QuestionSubmissionForm.tsx:66]: Do you have a strong reason to use a fragment here?

Comment (Akshay1018) [reply src/components/QuestionSubmissionForm.tsx:66]: Actually, we need fragments to return multiple elements. Here, I don't think so we need can wrap in the main div. We can place the Transition code exactly after `</form>.`

Comment (subhoghoshX) [reply src/components/QuestionSubmissionForm.tsx]: Nothing wrong with using a fragment, it's just it was shifting the whole html and the diff looked way more complicated than it should be. Now it looks good. Thanks :) Keeping the diff to a minimum makes the reviewer's life a little easier :smile:

Comment (Akshay1018) [reply src/components/QuestionSubmissionForm.tsx]: I appreciate it.

File: src/components/QuestionSubmissionForm.tsx [hunk 3]
```diff
@@ -1,4 +1,8 @@
+import React, { useState, Fragment } from 'react';
 import { type SubmitHandler, useForm, FieldValues } from 'react-hook-form';
+
+//3rd party components
```
Comment (ykdojo) [inline src/components/QuestionSubmissionForm.tsx:4]:
```suggestion
// 3rd party components
```

**Review by ykdojo — APPROVED** [review]

**Review by subhoghoshX — APPROVED** [review]

**Pull request merged**